internal/pkg/upload/doc.go

```diff
@@ -0,0 +1,14 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+/*
+
+	This package concentrates the responsibility of following the
+	File Storage specification:
+	https://docs.google.com/document/d/1BlCQYxHwgGZMxysc0BiWxT9ZCLKwgryhkD9ryDNDIuI/edit?usp=sharing
```

Comment:
This file is not publicly accessible, please summarize what the upload package is for instead.

Does this module upload files to ES?